### PR TITLE
[WIP] Angle workaround in ps_border_corner shader

### PR DESCRIPTION
**Pull request by kvark**
r? @glennw

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/1311)

### CODE REVIEW

**Comment by kvark** [comment]
If confirmed to fix the issue with shader linking, I'll add it to the wiki.

File: webrender/res/ps_border_corner.vs.glsl [hunk 1]
```diff
@@ -104,15 +106,16 @@ int select_style(int color_select, vec2 fstyle) {
...
                               style.y == BORDER_STYLE_DASHED;
-            if (style.x != style.y && (has_dots || has_dashes))
-                return BORDER_STYLE_SOLID;
-            return style.x;
+            if (style.x == style.y || !(has_dots || has_dashes))
+                result = style.x;
         }
         case SIDE_FIRST:
-            return style.x;
+            result = style.x;
```
Comment (emilio) [inline webrender/res/ps_border_corner.vs.glsl]: need to break, here and above, right?

Comment (kvark) [inline webrender/res/ps_border_corner.vs.glsl]: Yes, I missed that. Thanks!

**Comment by msreckovic** [comment]
The error is still there:
Failed to link shader program: "ps_border_corner"
C:\fakepath(2686,1-9): error X3004: undeclared identifier '_has_dots'

Warning: D3D shader compilation failed with default flags. (vs_5_0)
 Retrying with skip validation
C:\fakepath(2686,1-9): error X3004: undeclared identifier '_has_dots'

Warning: D3D shader compilation failed with skip validation flags. (vs_5_0)
 Retrying with skip optimization
C:\fakepath(2686,1-9): error X3004: undeclared identifier '_has_dots'

Warning: D3D shader compilation failed with skip optimization flags. (vs_5_0)

**Comment by glennw** [comment]
@kvark What's the status of this?

**Comment by kvark** [comment]
@glennw ah, sorry, it's still actual. I just didn't want to waste Milan's time in testing the changes and was secretly hoping that someone on Windows would pick it up. I'll play with it tomorrow.

**Comment by kvark** [comment]
I wasn't able to get Angle building here at home, but I pushed a better version of the fix now, in case @msreckovic gets another chance to run it.

**Comment by glennw** [comment]
Closing for now due to inactivity. We can re-open if needed.

**Pull request closed without merging**